tests/test_conversation_utils.py

```diff
@@ -0,0 +1,75 @@
+from khoj.processor.conversation import utils
+from langchain.schema import ChatMessage
+import factory
+import tiktoken
+
+
+class ChatMessageFactory(factory.Factory):
+    class Meta:
+        model = ChatMessage
+
+    content = factory.Faker("paragraph")
+    role = factory.Faker("name")
+
+
+class TestTruncateMessage:
+    max_prompt_size = 4096
+    model_name = "gpt-3.5-turbo"
+    encoder = tiktoken.encoding_for_model(model_name)
+
+    def test_truncate_message_all_small(self):
+        chat_messages = ChatMessageFactory.build_batch(500)
+        assert len(chat_messages) == 500
+        tokens = sum([len(self.encoder.encode(message.content)) for message in chat_messages])
+        assert tokens > self.max_prompt_size
+
+        prompt = utils.truncate_message(chat_messages, self.max_prompt_size, self.model_name)
```

Comment:
nit: variable name isn't descriptive
```suggestion
        truncated_messages = utils.truncate_message(chat_messages, self.max_prompt_size, self.model_name)
```